### PR TITLE
Cleanup duplicated watches in gardenlet (1)

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area scalability
/kind cleanup

**What this PR does / why we need it**:

This PR cleans up usages of the generated informers (client-go style) in gardenlet's ControllerInstallation and Seed controllers.
It replaces them with informers from the controller-runtime cache to prevent duplicated watches if the `CachedRuntimeClients` feature gate is enabled.
To ease the transition, the PR first refactors the controllers to the `reconcile.Reconciler` interface.

(similar to https://github.com/gardener/gardener/pull/3658 ff., https://github.com/gardener/gardener/pull/3746 ff.)

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/2822, https://github.com/gardener/gardener/issues/4251, https://github.com/gardener/gardener/issues/1648

**Special notes for your reviewer**:

The PR includes some more cleanups and refactorings along the way (see dedicated commits and commit messages):
- rework the seed lease controller
- rework the seed controller unit tests to make them simpler and use a fake client instead of mocks
- unify some client/clientset parameters in favor of c-r clients

There will be one more PR performing the same refactorings/cleanups for the last big chunk of code: the shoot controller and operation package.

/squash

**Release note**:

```other operator
The Kubernetes version requirement for Seeds, that was raised to `>= 1.18` in [gardener@v1.28.0](https://github.com/gardener/gardener/releases/tag/v1.28.0), is now technically enforced as well.
```

### CODE REVIEW

File: pkg/operation/common/utils.go [hunk 1]
```diff
@@ -106,50 +107,39 @@ func GenerateAddonConfig(values map[string]interface{}, enabled bool) map[string
...
-	}
+func DeleteHvpa(ctx context.Context, c client.Client, namespace string) error {
+	labelSelectorHVPA := client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleHvpa}
 
 	// Delete all CRDs with label "gardener.cloud/role=hvpa"
 	// Workaround: Due to https://github.com/gardener/gardener/issues/2257, we first list the HVPA CRDs and then remove
 	// them one by one.
-	crdList, err := k8sClient.APIExtension().ApiextensionsV1beta1().CustomResourceDefinitions().List(ctx, listOptions)
-	if err != nil {
+	crdList := &apiextensionsv1beta1.CustomResourceDefinitionList{}
```
Comment (rfranzke) [inline pkg/operation/common/utils.go]: You could use `v1` now that the min Kubernetes version for seeds was raised to `1.18`.

**Review by rfranzke — COMMENTED** [review]
As usual, well done! :)

File: pkg/operation/common/utils.go (hunk 1 above)
Comment (timebertt) [reply pkg/operation/common/utils.go]: Yep, done.
I also bumped the [technical seed version requirement](https://github.com/gardener/gardener/blob/4e518bb440525fb2adc5796d70c4b0003c2c5f02/pkg/operation/seed/seed.go#L1395) to 1.18 in order to reflect the doc change in https://github.com/gardener/gardener/pull/4426.
cc @timuthy

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by timebertt** [comment]
Do we want to have a second opinion on this PR or should we continue with the next one?

**Comment by rfranzke** [comment]
I'm good, not sure if somebody else plans to take a look, but given that there wasn't too much reaction yet, I'm fine with proceeding.

**Comment by timebertt** [comment]
Ok, then let's merge this one now, so that we can include both PRs in the next release and validate them early on (cc @BeckerMax).

**Pull request merged**